pkg/guardian/guardian_test.go

```diff
@@ -103,6 +112,7 @@ func TestBasicFunctionality(t *testing.T) {
 	}()
 	defer func() {
 		close(stop)
+		time.Sleep(50 * time.Millisecond) // give the redisConfStore.RunSync() goroutine time to exit
```

Comment:
ehhhhhhh lol. Not a big fan of sleep and pray things happen in time. Not enough to request changes, just wanted to call it out.

Reply:
I agree. :) The test failures I was getting seemed mysterious (this was months ago so it's a bit foggy to me ATM) and probably due to a race. Which was confirmed by the "fix" of adding a sleep.